apache/superset
# feat: Labeled Error-bound Input

### SUMMARY
I have created a new dynamic error-bound input. When the user interacts with this form, if an error occurs - designated by a function passed into the `validationMethods` parameter - the error will replace any existing helper text until the error is resolved. `<LabeledErrorBoundInput />` accepts the following parameters in order to customize the input:
```label?: string;
  name: string;
  validationMethods:
    | { onBlur: (value: any) => void }
    | { onChange: (value: any) => void };
  errorMessage: string | null;
  helpText?: string;
  value: string | number | readonly string[] | undefined;
  required?: boolean;
  placeholder?: string;
  autocomplete?: string;
  type?: string;
  id?: string;
```

### ANIMATED GIF
![labeledErrorBoundInput](https://user-images.githubusercontent.com/55605634/118211275-899b8500-b431-11eb-9060-9bf522dc0a88.gif)


### TEST PLAN
- Run Storybook
- Click LabeledErrorBoundInput > Interactive LabeledErrorBoundInput
- Type success into the text box to observe error behavior

### ADDITIONAL INFORMATION
